From 381175792f1d3b7ef55cea40ec4f0d48f501e9a4 Mon Sep 17 00:00:00 2001
From: Martin Negyokru <negyokru@inf.u-szeged.hu>
Date: Fri, 28 Apr 2023 17:37:24 +0200
Subject: [PATCH] Fix devtools context menus

Set devtools custom menus on 'OnLoadCompleted' event.
This event should be triggered by the Js frontend
not by WebContentsObserver.

Fixes: QTBUG-113251
Change-Id: I815d818c5867d0f1d160a962be5a41a762699e00
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
(cherry picked from commit 1243b80433d4a18bc57cf43ab29d02e6007261f3)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---

diff --git a/src/core/devtools_frontend_qt.cpp b/src/core/devtools_frontend_qt.cpp
index 731414d..4ef8b35 100644
--- a/src/core/devtools_frontend_qt.cpp
+++ b/src/core/devtools_frontend_qt.cpp
@@ -162,7 +162,7 @@
     web_contents()->Focus();
 }
 
-void DevToolsFrontendQt::InspectElementCompleted()
+void DevToolsFrontendQt::OnLoadCompleted()
 {
     m_bindings->CallClientMethod("DevToolsAPI", "setUseSoftMenu", base::Value(true));
 }
diff --git a/src/core/devtools_frontend_qt.h b/src/core/devtools_frontend_qt.h
index 3b722c9..a4f1726 100644
--- a/src/core/devtools_frontend_qt.h
+++ b/src/core/devtools_frontend_qt.h
@@ -54,18 +54,18 @@
 
     // DevToolsUIBindings::Delegate overrides
     void ActivateWindow() override;
-    void InspectElementCompleted() override;
     void SetEyeDropperActive(bool active) override;
     void OpenInNewTab(const std::string &url) override;
     void InspectedContentsClosing() override;
+    void OnLoadCompleted() override;
 
+    void InspectElementCompleted() override{};
     void CloseWindow() override{};
     void Inspect(scoped_refptr<content::DevToolsAgentHost>) override{};
     void SetInspectedPageBounds(const gfx::Rect &) override{};
     void SetIsDocked(bool) override{};
     void SetWhitelistedShortcuts(const std::string &) override{};
     void OpenNodeFrontend() override{};
-    void OnLoadCompleted() override{};
     void ReadyForTest() override{};
     void ConnectionReady() override{};
     void SetOpenNewWindowForPopups(bool) override{};
